workflow/parser.py

```diff
@@ -0,0 +1,67 @@
+import os
+import subprocess
+
+
+class Parser:
+    def __init__(self, params):
+        self.params = params
+
+    def parse(self):
+
+        '''
+        - parse input config file 
+        - build TPS call 
+        - call TPS 
+        - gather outputs 
+        - return:
+            - list of outputs 
+            - output_folder
+            - out_label
+
+      
+        '''
+        # # grab TPS self.params 
+        build = []
+        OUT_FOLDER = os.path.dirname(os.path.dirname(os.path.abspath(__file__)))
+        OUT_LABEL = ""
+
+        # add TPS run call as first arg
+        build.extend(["bash", "./scripts/run"])
+
+        # grab self.params from config
+        req = self.params[1]["TPS"][0]["required"]
+        op = self.params[1]["TPS"][1]["optional"]
+
+        # filter given args
+        filtered = {**req, **op}
+        args = {key:val for key, val in filtered.items() 
+                if filtered[key] != ""}
+
+                    
+        for key, val in args.items():
+            if key == "outlabel":
```

Comment:
Do we need to check for outlabel and outfolder every time through the loop?  Can you check for those keys once outside the loop instead?

Reply:
I can now index the args dictionary directly